src/components/PopoverMenu/BasePopoverMenu.js

```diff
@@ -26,13 +29,64 @@ const defaultProps = {
 };
 
 class BasePopoverMenu extends PureComponent {
+    constructor(props) {
+        super(props);
+        this.state = {
+            focusedIndex: -1,
+        };
+        this.updateFocusedIndex = this.updateFocusedIndex.bind(this);
+        this.removeKeyboardShorcuts = this.removeKeyboardShorcuts.bind(this);
+        this.onModalHide = this.onModalHide.bind(this);
+    }
+
+    componentDidMount() {
+        if (!this.props.enableArrowKeysActions && !this.props.isVisible) {
+            return;
+        }
+
+        const shortcutConfig = CONST.KEYBOARD_SHORTCUTS.ENTER;
+        this.unsubscribeEnterKey = KeyboardShortcut.subscribe(shortcutConfig.shortcutKey, () => {
+            if (this.state.focusedIndex === -1) {
+                return;
+            }
+            this.props.onItemSelected(this.props.menuItems[this.state.focusedIndex]);
+            this.updateFocusedIndex(-1); // Reset the focusedIndex on selecting any menu
+        }, shortcutConfig.descriptionKey, shortcutConfig.modifiers, true);
+    }
+
+    componentWillUnmount() {
+        this.removeKeyboardShorcuts();
+    }
+
+    onModalHide() {
+        this.updateFocusedIndex(-1); // Reset the focusedIndex on modal hide
+        this.props.onMenuHide();
+    }
+
+    removeKeyboardShorcuts() {
```

Comment:
I think that this method isn't needed. It's only used once so just inline it in `componentWillUnmount`

Reply:
Removed.